### DISCUSSION TITLE
Can't push my repo to github

### DISCUSSION THREAD

**Discussion by dizzzBM** [community/community#29521] [Other Feature Feedback, Questions, & Ideas]
Hi Guys please help me, i can't push my github repo (this repo from gitlab before, and i wanted to migrate it to github) because i have one file who have a size of more than 100M. I'm already tired git lfs but still getting the same issue also tired to remove the file but still getting same error, anyone can help me please?

remote: Resolving deltas: 100% (19631/19631), done.
remote: error: Trace: 916547bd45c43b1b9f47acc41cd5dc6c6966939da5cb88b3595a26aa716b9386
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File terraform/.terraform/plugins/darwin_amd64/terraform-provider-aws_v2.65.0_x4 is 177.03 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com/.

**Comment by airtower-luna** [comment]
You need to remove the file not only from the latest commit but from all of the repository history, old commits and their files still get pushed. Some ways to do that are described in [Removing sensitive data from a repository](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository), what works to fully purge a secret file also works for a big file. If you want to learn about editing history in general see [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

**Reply by dizzzBM** [reply]
@airtower-luna thanks for the answer and let me try your method. thanks mate

**Comment by codesandbax** [comment]
#28859 #29593 #28677 #29593 @airtower-luna
